File: binary/Binary.java
package binary;

public class Binary {
    public static void main(String[] args) {
        int[]    nums = { 1, 2, 4, 5, 6, 7, 8, 12, 14, 15 };
        System.out.println(binarySearch(nums, 12));
    }

    static int binarySearch(int[] nums, int target) {
        int start = 0;
        int end = nums.length - 1;
        int middle = Math.round((start + end) / 2);
        while (start < end) {
            if (target < nums[middle]) {
                end = middle - 1;

            } else {
                start = middle + 1;
            }
            middle = Math.round((start + end) / 2);
        }
        return target == nums[middle] ? middle : -1;
    }
}

File: binary.js
let array = [0, 2, 4, 5, 7, 8, 10, 12, 23]


let binarySearch = (nums, target) => {
    let start = 0
    let end = nums.length - 1
    let middle = Math.floor((start + end) / 2)
    console.log(middle)

    while (start <= end) {
        if (target < nums[middle]) {
            console.log(`target value is less: ${target} < ${nums[middle]}`);
            end = middle - 1
            console.log(`end: ${end} = ${middle} - 1`);
        }
        else {
            console.log(`target value is more: ${target} > ${nums[middle]}`);
            start = middle + 1
            console.log(`start: ${end} = ${middle} + 1`);
        }
        middle = Math.floor((start + end) / 2)
        console.log(`middle: ${middle} = ${start} + ${end} / 2`);
    }
    return array[middle] === target ? `found at index ${middle} -> ${target}` : `${start} : Ceiling Index : value ${nums[start]}`
}

console.log(binarySearch(array, 14))
